python/cudf/cudf/core/column/column.py

```diff
@@ -2984,6 +2996,14 @@ def as_column(
                 arbitrary,
                 from_pandas=True,
             )
+            if (
+                cudf.get_option("mode.pandas_compatible")
+                and inferred_dtype == "mixed"
+                and not isinstance(
+                    pyarrow_array.type, (pa.ListType, pa.StructType)
```

Comment:
Is this because pyarrow coerced mixed data to the same type with its inference? Like `[1, "1"]` coercing to all `int` or similar?